### DISCUSSION TITLE
InMemory cannot implement Lcobucci

### DISCUSSION THREAD

**Discussion by betysam55** [lcobucci/jwt#782] [Q&A] [answered]
`{
    "message": "Lcobucci\\JWT\\Signer\\Key\\InMemory cannot implement Lcobucci\\JWT\\Signer\\Key - it is not an interface",
"exception": "Symfony\\Component\\Debug\\Exception\\FatalErrorException",
    "file": "/Users/sam/Desktop/easy/easy-ethiopia-web/vendor/lcobucci/jwt/src/Signer/Key/InMemory.php",
    "line": 14,
    "trace": []
}`

can any one  help

**Comment by lcobucci** [comment]
@betysam55 which version of `lcobucci/jwt` you have installed and how did you install it?

**Comment by la-jamesh** [comment]
Seeing a similar issue with Laravel 8 and passport 10.2

**Reply by lcobucci** [reply]
And what version of lcobucci/jwt you have?

**Reply by la-jamesh** [reply]
```
root@87a0d92c34d9:/var/www# composer show lcobucci/jwt
name     : lcobucci/jwt
descrip. : A simple library to work with JSON Web Token and JSON Web Signature
keywords : JWS, jwt
versions : * 4.1.5
```

**Reply by la-jamesh** [reply]
Trying to upgrade from PHP 7 to PHP 8. Is how we stumbled across this.

**Reply by lcobucci** [reply]
And that same message is happening? Is it possible that you have version 3.4 also installed?

**Reply by lcobucci** [reply]
Perhaps opcache?

**Reply by francis-roux-serret** [reply]
any update ?

**Reply by farmani** [reply]
one of your dependencies rely on older version of lcobucci/jwt

**Comment by lcobucci** [comment, marked as answer]
As we found out in https://github.com/lcobucci/jwt/issues/838 the Auth0 SDK v7 (https://github.com/auth0/auth0-PHP/blob/7.x/composer.json#L17) requires a library which is a fork of an old an unmaintained version of this library to provide support to PHP 8 (and it seems like the `conflicts` section was not updated on that fork).

The v8 of that SDK solves the incompatibility by removing dependencies.
